### PR TITLE
cmake: Remove DEMOS_WSI_SELECTION

### PR DESCRIPTION
**Pull request by juan-lunarg**
DEMOS_WSI_SELECTION does nothing.

### CODE REVIEW

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM.  This sure looks like it has been a fossil for a very long time.

**Comment by juan-lunarg** [comment]
> LGTM. This sure looks like it has been a fossil for a very long time.

Looks like it was copy pasted from [VkCube](https://github.com/KhronosGroup/Vulkan-Tools/blob/6e5bd86b76c092fcd5d8a873260a8a96f43705d5/cube/CMakeLists.txt#L57)

**Comment by Tony-LunarG** [comment]
I'd bet it's from a time when cube and validation layers shared a repo

File: CMakeLists.txt [hunk 1]
```diff
@@ -141,7 +141,6 @@ if (UNIX AND NOT APPLE) # Options for Linux only
     option(BUILD_WSI_XCB_SUPPORT "Build XCB WSI support" ON)
     option(BUILD_WSI_XLIB_SUPPORT "Build Xlib WSI support" ON)
     option(BUILD_WSI_WAYLAND_SUPPORT "Build Wayland WSI support" ON)
-    set(DEMOS_WSI_SELECTION "XCB" CACHE STRING "Select WSI target for demos (XCB, XLIB, WAYLAND, DISPLAY)")
```
Comment (ncesario-lunarg) [inline CMakeLists.txt:144]: :confused:

**Review by ncesario-lunarg — APPROVED** [review]

**Pull request merged**